apps/condo/domains/ticket/utils/serverSchema/TicketDocumentGeneration/paidWorks.js

```diff
@@ -0,0 +1,220 @@
+        sortBy: ['createdAt_ASC'],
+    })
+
+    const currencyCode = get(invoices, '0.currencyCode') || DEFAULT_INVOICE_CURRENCY_CODE
+    let totalSum = Big(0), totalVAT = Big(0)
+
+    const listOfWorks = invoices.reduce((acc, invoice) => {
+        const rows = Array.isArray(invoice.rows) ? invoice.rows : []
+        acc.push(...rows.map((row, index) => {
+            let price
+            if (row.isMin) {
+                price = Big(0)
+            } else {
+                price = !Number.isNaN(parseFloat(row.toPay)) ? Big(row.toPay) : Big(0)
+            }
+            const count = !Number.isNaN(parseFloat(row.count)) ? Big(row.count) : Big(1)
+            const vatPercent = !Number.isNaN(parseFloat(row.vatPercent)) ? Big(get(row, 'vatPercent', 0)) : Big(0)
+
+            const sum = price.times(count)
+
+            totalSum = totalSum.plus(sum) 
+            totalVAT = totalVAT.plus(sum.times(vatPercent).div(100))
+
+            return {
+                number: index + 1,
+                name: row.name,
+                count: String(count),
+                price: renderMoney(price, currencyCode, locale),
+                vat: String(vatPercent),
+                sum: renderMoney(sum, currencyCode, locale),
+            }
+        }))
+
+        return acc
+    }, [])
+    
+    // NOTE: This doc needs two variables to be generated:
+    // 1. Values that will go into the table (documentTextData)
+    // 2. Values that will be embeded into the text of the document (documentDataInTable)
+
+    const documentTextData = {
+        header: {
+            generalDate: printDate.format(getDataFormatByLocale(locale)),
+        },
+        company: {
+            name: get(organization, 'name'),
+            psrn: psrn,
+            tin: get(organization, 'tin'),
+            iec: iec,
+            address: organizationAddress,
+            phone: get(organization, 'phone'),
+        },
+        bankDetails: {
+            accountNumber: get(invoices, '0.accountNumber'),
+            bankName: get(invoices, '0.recipient.bankName'),
+            bankAccount: get(invoices, '0.recipient.bankAccount'),
+            bic: get(invoices, '0.recipient.bic'),
+        },
+        totalInWords: {
+            totalSum: moneyToWords(totalSum, { locale, currencyCode }),
+            totalVAT: moneyToWords(totalVAT, { locale, currencyCode }),
+        },
+    }
+
+    const documentDataInTable = {
+        client: {
+            name: get(contact, 'name'),
+        },
+        listOfWorks,
+        totalInNumbers: {
+            totalSum: renderMoney(totalSum, currencyCode, locale),
+            totalVAT: renderMoney(totalVAT, currencyCode, locale),
+        },
+        executor: {
+            name: get(employee, 'name'),
+        },
+    }
+
+    const documentData = {
+        ...replaceAllUndefinedValues(documentTextData, _changeToBlanks),
+        ...replaceAllUndefinedValues(documentDataInTable, _changeToEmptyCell),
+    }
```

Comment:
I really don't understand why these transformations are needed...
Usually just use inline default values, it's much simpler

Reply:
Why exactly undefined? That the value will be null or empty line?